File: integrations/src/main/java/framework/core/logger/ThirdPartyRequestResponseLoggerImpl.java
package framework.core.logger;

import framework.config.BusinessContext;
import framework.core.data.ApiContext;
import framework.core.entity.ThirdPartyResponseLog;
import framework.core.repository.ThirdPartyResponseLogRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpMethod;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
public class ThirdPartyRequestResponseLoggerImpl implements ThirdPartyRequestResponseLogger {

    private final ThirdPartyResponseLogRepository thirdPartyResponseLogRepository;

    @Autowired
    public ThirdPartyRequestResponseLoggerImpl(ThirdPartyResponseLogRepository thirdPartyResponseLogRepository) {
        this.thirdPartyResponseLogRepository = thirdPartyResponseLogRepository;
    }

    @Override
    public UUID registerRequest(
            BusinessContext businessContext,
            ApiContext apiContext,
            HttpMethod requestMethod,
            String url,
            String requestBody) {
        ThirdPartyResponseLog logEntry = new ThirdPartyResponseLog();
        logEntry.setEntityType(1); // Default entity type
        logEntry.setEntityId(businessContext.getEntityId());
        logEntry.setRequestMethod(requestMethod.toString());
        logEntry.setUrl(url);
        logEntry.setRequest(requestBody);
        logEntry.setBusinessPurpose(businessContext.getBusinessPurpose());
        logEntry.setBusinessEntityName(businessContext.getEntityName());
        logEntry.setApiPurpose(apiContext.getApiPurpose());

        ThirdPartyResponseLog savedLog = thirdPartyResponseLogRepository.save(logEntry);
        return savedLog.getId() != null ? savedLog.getId() : UUID.randomUUID();
    }

    @Override
    public UUID registerThirdPartyRequest(
            BusinessContext businessContext,
            ApiContext apiContext,
            HttpMethod requestMethod,
            String url,
            String requestBody) {
        // For now, this is the same as registerRequest
        return registerRequest(businessContext, apiContext, requestMethod, url, requestBody);
    }

    @Override
    public void registerResponse(
            UUID thirdPartyRequestId,
            String responseBody,
            long responseTimeInMs,
            int requestStatus) {
        var logEntry = thirdPartyResponseLogRepository.findById(thirdPartyRequestId).orElse(null);
        if (logEntry != null) {
            logEntry.setResponse(responseBody);
            logEntry.setResponseTimeInMs(responseTimeInMs);
            logEntry.setHttpStatusCode(requestStatus);
            thirdPartyResponseLogRepository.save(logEntry);
        }
    }
}

File: integrations/src/main/kotlin/framework/core/logger/ThirdPartyRequestResponseLoggerImpl.kt
package framework.core.logger

import framework.config.BusinessContext
import framework.core.data.ApiContext
import framework.core.entity.ThirdPartyResponseLog
import framework.core.repository.ThirdPartyResponseLogRepository
import org.springframework.http.HttpMethod
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class ThirdPartyRequestResponseLoggerImpl(
    private val thirdPartyResponseLogRepository: ThirdPartyResponseLogRepository
) : ThirdPartyRequestResponseLogger {

    override fun registerRequest(
        businessContext: BusinessContext,
        apiContext: ApiContext,
        requestMethod: HttpMethod,
        url: String,
        requestBody: String?
    ): UUID {
        val logEntry = ThirdPartyResponseLog(
            entityType = 1, // Default entity type
            entityId = businessContext.entityId,
            requestMethod = requestMethod.toString(),
            url = url,
            request = requestBody,
            businessPurpose = businessContext.businessPurpose,
            businessEntityName = businessContext.entityName,
            apiPurpose = apiContext.apiPurpose
        )

        val savedLog = thirdPartyResponseLogRepository.save(logEntry)
        return savedLog.id ?: UUID.randomUUID()
    }

    override fun registerThirdPartyRequest(
        businessContext: BusinessContext,
        apiContext: ApiContext,
        requestMethod: HttpMethod,
        url: String,
        requestBody: String?
    ): UUID {
        // For now, this is the same as registerRequest
        return registerRequest(businessContext, apiContext, requestMethod, url, requestBody)
    }

    override fun registerResponse(
        thirdPartyRequestId: UUID,
        responseBody: String,
        responseTimeInMs: Long,
        requestStatus: Int
    ) {
        val logEntry = thirdPartyResponseLogRepository.findById(thirdPartyRequestId).orElse(null)
        if (logEntry != null) {
            logEntry.response = responseBody
            logEntry.responseTimeInMs = responseTimeInMs
            logEntry.httpStatusCode = requestStatus
            thirdPartyResponseLogRepository.save(logEntry)
        }
    }
}
